python/src/scipp/plot/model.py

```diff
@@ -193,87 +191,31 @@ def get_data_coord(self, name, dim):
         """
         return self.data_arrays[name].coords[dim]
 
-    def _select_bins(self, coord, dim, start, end):
-        """
-        Method to speed up the finding of bin ranges
-        """
-        bins = coord.shape[-1]
-        if len(coord.dims) != 1:  # TODO find combined min/max
-            return dim, slice(0, bins - 1)
-        # scipp treats bins as closed on left and open on right: [left, right)
-        first = sc.sum(coord <= start, dim).value - 1
-        last = bins - sc.sum(coord > end, dim).value
-        if first >= last:  # TODO better handling for decreasing
-            return dim, slice(0, bins - 1)
-        first = max(0, first)
-        last = min(bins - 1, last)
-        return dim, slice(first, last + 1)
-
-    def resample_data(self, array, rebin_edges):
-        """
-        Resample a DataArray according to new bin edges.
-        """
-        dslice = array
-        # Select bins to speed up rebinning
-        for dim in rebin_edges:
-            this_slice = self._select_bins(array.coords[dim], dim,
-                                           rebin_edges[dim][dim, 0],
-                                           rebin_edges[dim][dim, -1])
-            dslice = dslice[this_slice]
```

Comment:
Did this optimization disappear and get lost, or did it move somewhere else?

Edit: I see it now in `model2d.py`. Is this only relevant in the 2d case?

Reply:
Yes i think so. We only use this for dynamic image resampling.
We do use rebin when we slice in 1d and 3d also, but then we are rebinning to the entire range:
https://github.com/scipp/scipp/blob/cleanup_slider_interface/python/src/scipp/plot/model.py#L214